src/lib/components/ReactSearchKit/ReactSearchKit.js

```diff
@@ -18,17 +18,30 @@ import { buildUID } from '../../util';
 export class ReactSearchKit extends Component {
   constructor(props) {
     super(props);
+
+    const validate = (initialQueryState) => {
+      const layoutOptions = ['list', 'grid'];
+      if (
+        initialQueryState['layout'] &&
+        !layoutOptions.includes(initialQueryState['layout'])
+      ) {
+        throw new Error('Layout option is invalid.');
+      }
+
+      return initialQueryState;
```

Comment:
Sorry, I was not extremely clear... :( it would be great also if you pass an extra key, it fails.
e.g. `initialQueryState={'hello': 'world', ...}` should fail because not a valid key for the state.

pseudocode:
```
for key in initalQueryState:
  if key not in INITIAL_STATE.keys():
       throw...
```

other validations maybe: page should be a number, anything else?